azurerm/resource_arm_operational_insight_workspace.go

```diff
@@ -0,0 +1,224 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+
+	"github.com/Azure/azure-sdk-for-go/arm/operationalinsights"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmOperationalInsightWorkspaceService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmOperationalInsightWorkspaceCreateUpdate,
+		Read:   resourceArmOperationalInsightWorkspaceRead,
+		Update: resourceArmOperationalInsightWorkspaceCreateUpdate,
+		Delete: resourceArmOperationalInsightWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRmOperationalInsightWorkspaceName,
+			},
+			"location": {
+				Type:      schema.TypeString,
+				Required:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
```

Comment:
can we swap location over to using the locationSchema? i.e. `"location": locationSchema()`

Reply:
Done